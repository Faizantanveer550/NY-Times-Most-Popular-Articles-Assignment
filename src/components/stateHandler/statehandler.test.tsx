import { render, fireEvent, screen } from "@testing-library/react";
import StateHandler from "./stateHandler";

const errorMock = {
  formattedMsg: "Error message",
};

test("Snapshot Test", () => {
  const { asFragment } = render(
    <StateHandler error={errorMock} isLoading={false}>
      <div />
    </StateHandler>
  );
  expect(asFragment()).toMatchSnapshot();
});

test("StateHandler renders error message and close button", () => {
  render(
    <StateHandler error={errorMock} isLoading={false}>
      <div />
    </StateHandler>
  );

  expect(screen.getByText(errorMock.formattedMsg)).toBeInTheDocument();
  expect(screen.getByRole("button")).toBeInTheDocument();
});

test("StateHandler hides error message on close button click", () => {
  render(
    <StateHandler error={errorMock} isLoading={false}>
      <div />
    </StateHandler>
  );

  const closeButton = screen.getByTestId("state-handler-error-close-btn");
  fireEvent.click(closeButton);

  expect(screen.queryByText(errorMock.formattedMsg)).not.toBeInTheDocument();
});

test("StateHandler renders custom blank slate if provided", () => {
  const customBlankSlate = <p>Loading...</p>;
  render(
    <StateHandler isLoading={true} blankSlate={customBlankSlate} error={false}>
      <div />
    </StateHandler>
  );

  expect(screen.getByText("Loading...")).toBeInTheDocument();
});

test("StateHandler renders children when no error or loading", () => {
  const children = <p>This is the content.</p>;
  render(
    <StateHandler isLoading={false} error={false}>
      {children}
    </StateHandler>
  );

  expect(screen.getByText("This is the content.")).toBeInTheDocument();
});

test("StateHandler renders default blank slate when not provided", () => {
  render(
    <StateHandler isLoading={true} error={false}>
      <div />
    </StateHandler>
  );

  expect(
    screen.getByTestId("state-handler-blank-slate-def-content")
  ).toBeInTheDocument();
});
