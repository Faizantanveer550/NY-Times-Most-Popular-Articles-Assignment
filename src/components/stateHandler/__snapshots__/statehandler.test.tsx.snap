// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot Test 1`] = `
<DocumentFragment>
  <div
    class="sc-aYaIB fLrPkl simple-component-error-alert"
  >
    <span>
      Error message
    </span>
    <button
      data-testid="state-handler-error-close-btn"
    >
      <svg
        aria-hidden="true"
        class="svg-inline--fa fa-times faW-11"
        data-icon="times"
        data-prefix="fas"
        focusable="false"
        height="27px"
        role="img"
        viewBox="0 0 352 512"
        width="21px"
      >
        <path
          d="M242.72 256l100.07-100.07c12.28-12.28 12.28-32.19 0-44.48l-22.24-22.24c-12.28-12.28-32.19-12.28-44.48 0L176 189.28 75.93 89.21c-12.28-12.28-32.19-12.28-44.48 0L9.21 111.45c-12.28 12.28-12.28 32.19 0 44.48L109.28 256 9.21 356.07c-12.28 12.28-12.28 32.19 0 44.48l22.24 22.24c12.28 12.28 32.2 12.28 44.48 0L176 322.72l100.07 100.07c12.28 12.28 32.2 12.28 44.48 0l22.24-22.24c12.28-12.28 12.28-32.19 0-44.48L242.72 256z"
          fill="#a8a8a8"
        />
      </svg>
    </button>
  </div>
</DocumentFragment>
`;
